dvc/render/match.py

```diff
@@ -94,13 +108,24 @@ def match_defs_renderers(
 
             converter = _get_converter(renderer_cls, inner_id, props, definitions_data)
 
-            dps, rev_props = converter.flat_datapoints(rev)
+            try:
+                dps, rev_props = converter.flat_datapoints(rev)
+            except Exception as e:  # noqa: BLE001, pylint: disable=broad-except
+                def_errors[rev] = e
```

Comment:
It looks like there were no tests added to cover that we handle these `definition_errors`

Reply:
It's always hard to write a good test in these cases. What would be a good candidate to test for `flat_datapoints()`?

Reply:
Add a test in `tests/integrations` that checks that the `definition_errors` errors are included.

This unit test checks has example of what kind of input data raise the exceptions:

https://github.com/iterative/dvc/blob/121d56b7391f5325e5eb6ff41448980f0ede00df/tests/unit/render/test_vega_converter.py#L468-L471